### ISSUE TITLE
guru meditation error when streaming

### ISSUE THREAD

**Issue by aminty** [jameszah/ESP32-CAM-Video-Recorder-junior#25]
Hi mr James.
I hope you’re doing well.

around 3 week , i'm confusing about a problem.
in your source code(recorder junior v58) with easy install and with arduino IDE after uploading, when i'm trying to streaming  , esp will rebooted.
and in serial monitor this line appear:
(guru meditation error: core 0 panic'ed (loadprohibited). exception was unhandled.)

But except for this option, all other option work well

**Comment by Roobyx** [comment]
Getting the exact same error here.

**Comment by jameszah** [comment]
Could you cut and paste the serial monitor with the series of messages, and the stack of numbers from the "panic" to help me locate the problem.  And the version number, and version number of compiler. libraries if you re-compiled the program.

**Comment by CasperJohansen** [comment]
Getting the same error repeatedly.
V58: https://pastebin.com/MjLBhQNv (only happens when trying to stream or snapshot, records fine to sd)
v60: https://pastebin.com/ErqZVwCb (happens immidiately)

Both installed from one-click installer, and severeal different sd-cards tried

**Comment by jameszah** [comment]
Your module only has 2GB of spi-ram.  The code is built for 4GB in most esp32-cam modules.  

You could squeeze the memory a little here ... to repeatedly malloc and free the memory as needed ... I wish I would have spent an extra 20 minutes on that 2 years ago.

https://github.com/jameszah/ESP32-CAM-Video-Recorder-junior/blob/976b276c28f2d28ec8ba44c9c43fa663b712f1c0/v60/ESP32-CAM-Video-Recorder-junior-60x.4.7/ESP32-CAM-Video-Recorder-junior-60x.4.7.ino#L2482

....

![image](https://github.com/jameszah/ESP32-CAM-Video-Recorder-junior/assets/36938190/646ec71c-187b-4d29-9ebd-bd95e7d263a4)

**Comment by duchengyao** [comment]
what size should be framebuffer in 2GB memory?

**Comment by Mostakim52** [comment]
I seem to be having the same problem. Is there a fix yet?

**Comment by jameszah** [comment]
The program is built for a standard esp32-cam module with 4mb of psram.

**Comment by v2e** [comment]
> what size of framebuffer should be in 2GB memory?

I tested with (512*192) like this:

```
framebuffer = (uint8_t*)ps_malloc(512 * 192);
framebuffer2 = (uint8_t*)ps_malloc(512 * 192);
framebuffer3 = (uint8_t*)ps_malloc(512 * 192);
```

And it seems to work (more or less) fine, and even the streaming and image series work correctly with OV2640 camera.
For the 1600x1200 mode (maximum possible for this camera), and the quality level of 12 (set by the software automatically), the statistics shows the average frame size  of about 71 kB. Meanwhile (512*192) is almost 100 kB (I guess, this is what the framebuffer is intended for), so there is some extra space just in case.

This may not be reliable, but at least it allows to get some videos from the 2MB ESP32-CAM's. I hope this information will be useful for other people too.

**Comment by jameszah** [comment]
You are right - some old code there - it is set up for the all frame sizes in bright color.

512kb is overly conservative - 128kb should handle most cases.
